sros/docker/launch.py

```diff
@@ -287,15 +287,18 @@ def _parse(cfg, obj, skip_nics=False):
     if "___" in cfg:
         variant["deployment_model"] = "distributed"
         variant["lcs"] = []
-        variant["max_nics"] = 0
+        # >> No need to recompute max_nics
+        # variant["max_nics"] = 0
         for hw_part in cfg.split("___"):
             if "cp: " in hw_part:
                 variant["cp"] = _parse(hw_part.strip(), None, skip_nics=True)
             elif "lc: " in hw_part:
                 lc = _parse(hw_part.strip(), None)
-                if "max_nics" in lc:
-                    variant["max_nics"] = variant["max_nics"] + int(lc["max_nics"])
-                    lc.pop("max_nics")
+                # >> No need to rempute max_nics
+                # if "max_nics" in lc:
+                #    variant["max_nics"] = variant["max_nics"] + int(lc["max_nics"])
+                    # Need to keep this information
+                    # lc.pop("max_nics")
```

Comment:
why is this removed? what would be the max nic count for a line card?

Reply:
refreshing myself on this code piece, I think the max_nics was possible to provide manually via the `type` definition, using something like this:

https://github.com/hellt/vrnetlab/blob/ce9e236b9370d71f04dae04f6ba6c78813b79760/sros/docker/launch.py#L253

now when I look at that, it seems that it makes sense to allow specifying `max_nics` in the `cp` portion of a free-formed type, not the `lc`.
That way you specify the amount of nics that this particular chassis is intended to have.

This way the commented code in that block is indeed not needed, but then we need to derive the max_nics from the `cp` portion

Reply:
> Hi @gusman thanks for contribution can you provide a topology that leverages multi-lc setup for me to test it?

I use this topology for testing.

```
name: nodes_sros_lc_test
prefix: __lab-name
topology:
  nodes:
    P1:
      kind: vr-sros
      image: vrnetlab/vr-sros:22.7.R2
      type: 'cp: cpu=2 min_ram=4 chassis=sr-7 slot=A card=cpm5 ___ lc: cpu=4 min_ram=4 max_nics=10 chassis=sr-7 slot=1 card=iom4-e mda/1=me6-10gb-sfp+ ___ lc: cpu=4 min_ram=4 max_nics=10 chassis=sr-7 slot=2 card=iom4-e mda/1=me6-10gb-sfp+'
      license: /license/timos-latest.txt
    P2:
      kind: vr-sros
      image: vrnetlab/vr-sros:22.7.R2
      type: 'cp: cpu=2 min_ram=4 chassis=sr-7 slot=A card=cpm5 ___ lc: cpu=4 min_ram=4 max_nics=10 chassis=sr-7 slot=1 card=iom4-e mda/1=me6-10gb-sfp+ ___ lc: cpu=4 min_ram=4 max_nics=10 chassis=sr-7 slot=2 card=iom4-e mda/1=me6-10gb-sfp+'
      license: /license/timos-latest.txt
    PE:
      kind: vr-sros
      image: vrnetlab/vr-sros:22.7.R2
      type: 'cp: cpu=2 min_ram=4 chassis=sr-7 slot=A card=cpm5 ___ lc: cpu=4 min_ram=4 max_nics=10 chassis=sr-7 slot=1 card=iom4-e mda/1=me6-10gb-sfp+ ___ lc: cpu=4 min_ram=4 max_nics=10 chassis=sr-7 slot=2 card=iom4-e mda/1=me6-10gb-sfp+'
      license: /license/timos-latest.txt
  links:
  - endpoints:
    - P1:eth11
    - P2:eth11
  - endpoints:
    - P1:eth1
    - PE:eth1
  - endpoints:
    - PE:eth11
    - P2:eth1
```

Reply:
> refreshing myself on this code piece, I think the max_nics was possible to provide manually via the `type` definition, using something like this:
> 
> https://github.com/hellt/vrnetlab/blob/ce9e236b9370d71f04dae04f6ba6c78813b79760/sros/docker/launch.py#L253
> 
> now when I look at that, it seems that it makes sense to allow specifying `max_nics` in the `cp` portion of a free-formed type, not the `lc`. That way you specify the amount of nics that this particular chassis is intended to have.

[g] Yes correct, but in my PR it's not accommodated yet

> 
> This way the commented code in that block is indeed not needed, but then we need to derive the max_nics from the `cp` portion

[g] Kindly correct my understanding, so it's possible that MDA to be attached on CP part and LC part, I mean on TIMOS line description.

Reply:
mm, I think what I said above is not correct...

Let's take your example, where you define your custom chassis with one CPM and two Line Card VMs:

```
type: 'cp: cpu=2 min_ram=4 chassis=sr-7 slot=A card=cpm5 ___ lc: cpu=4 min_ram=4 max_nics=10 chassis=sr-7 slot=1 card=iom4-e mda/1=me6-10gb-sfp+ ___ lc: cpu=4 min_ram=4 max_nics=10 chassis=sr-7 slot=2 card=iom4-e mda/1=me6-10gb-sfp+'
```

As a user of this lab you need to understand how many ports each VM has, so that you can understand which index to use in the `ethX` notation in the clab file.
Your first LC has 10 ports, so when you use `eth1..eth10` ports in your clab file you will effectively connect to the first line card VM

When you define a port `eth11` it is going to be the first port on a second line card VM, because first 10 ports are allocated for the 1st LC.

So I guess max_nics still should be used in every linecard definition, for you to derive the start index (and this is what you already do)

Reply:
Yes correct, if max_nics it's not derived in every linecard definition, then we need to build a reference data that contains number of ports on each linecard.